aten/src/ATen/native/vulkan/VulkanOpaqueTensorImpl.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <ATen/OpaqueTensorImpl.h>
+
+namespace at {
+// The only difference from OpaqueTensorImpl is faking strides(), stride(),
+// is_contiguous(). The main intention for this is to be able to run torchscript
+// model on Vulkan backend. Strides are not supported on Vulkan side, plan to
+// support them.
+template <typename OpaqueHandle>
+struct CAFFE2_API VulkanOpaqueTensorImpl
```

Comment:
CAFFE2_API should not be necessary considering that templates are to be both declared and defined in the header file, making the implementation available to downstream users in the header itself.